### ISSUE TITLE
Update to Go 1.16 and Node 14

### ISSUE THREAD

**Issue by loleg** [ProxeusApp/proxeus-core#217]
The version of Go used by this project is nearly 2 years old, and is a couple of versions behind the stable release (1.15.8) standard to most development operating systems. Furthermore, on attempting to produce a clean Docker build I encountered this issue:

```
build github.com/ProxeusApp/proxeus-core/main: cannot load hash/maphash: malformed module path "hash/maphash": missing dot in first path element
make: *** [Makefile:108: server-docker] Error 1
```

This appears to be related to [golang#40067](https://github.com/golang/go/issue/40067), and is solved by updating to use golang 1.16. Furthermore, in recent upgrade maintenance I have had to `replace` (freeze) several package versions, which may be no longer necessary. 

In summary: moving to the current stable Go release (1.15 or 1.16) seems like a sensible thing to do to keep this project up to date. While we are doing a major upgrade, we may as well update to the LTS version of Node.js (14.x) on the frontend as well, since Node.js 8.x LTS Carbon is deprecated and no longer actively supported.

**Comment by loleg** [comment]
Local end-to-end testing is looking good.

**Comment by loleg** [comment]
I noticed there are several warnings in the documentation about newer (12+) versions of Node, but have decided to pull the build up past it anyway. Automated tests are passing for me locally, and we will need to do end-to-end testing of the UI soon anyway.
